File: StudentOglasi/UI/studentoglasi_admin/lib/models/PrijavePraksa/prijave_praksa.dart

import 'package:json_annotation/json_annotation.dart';
import 'package:studentoglasi_admin/models/Praksa/praksa.dart';
import 'package:studentoglasi_admin/models/StatusPrijave/statusprijave.dart';
import 'package:studentoglasi_admin/models/Student/student.dart';

part 'prijave_praksa.g.dart';

@JsonSerializable()
class PrijavePraksa {
  int? studentId;
  int? praksaId;
  String? propratnoPismo;
  String? cv;
  String? certifikati;
  int? statusId;
  Praksa? praksa;
  StatusPrijave? status;
  Student? student;


  PrijavePraksa(this.studentId, this.praksaId, this.propratnoPismo,this.cv,this.certifikati,this.statusId,this.praksa,this.status,this.student);
  factory PrijavePraksa.fromJson(Map<String, dynamic> json) => _$PrijavePraksaFromJson(json);

  Map<String, dynamic> toJson() => _$PrijavePraksaToJson(this);
}
/* {
      "studentId": 1,
      "praksaId": 2,
      "propratnoPismo": "...",
      "cv": "..",
      "certifikati": "...",
      "statusId": 1,
      "praksa": null,
      "status": null,
      "student": null
    }*/

File: StudentOglasi/StudentOglasi.Services/Database/PrijavePraksa.cs
using System;
using System.Collections.Generic;

namespace StudentOglasi.Services.Database;

public partial class PrijavePraksa
{
    public int StudentId { get; set; }

    public int PraksaId { get; set; }

    public string? PropratnoPismo { get; set; }

    public string Cv { get; set; } = null!;

    public string? Certifikati { get; set; }

    public int StatusId { get; set; }

    public virtual Prakse Praksa { get; set; } = null!;

    public virtual StatusPrijave Status { get; set; } = null!;

    public virtual Studenti Student { get; set; } = null!;
}
